video/mp_image.c

```diff
@@ -525,6 +533,8 @@ void mp_image_copy_attributes(struct mp_image *dst, struct mp_image *src)
                 memcpy(dst->planes[1], src->planes[1], MP_PALETTE_SIZE);
         }
     }
+    if (src->icc_profile)
+        dst->icc_profile = av_buffer_ref(src->icc_profile);
```

Comment:
Doesn't properly unref or clear `dst->icc_profile`. Also lack of error handling.